demisto_sdk/commands/format/update_incidentfields.py

```diff
@@ -44,42 +45,53 @@ def run_format(self) -> int:
-                if alais_field:
-                    alias_field_file_path = alais_field.get('file_path')
-                    alias_orig_field, _ = get_dict_from_file(path=alias_field_file_path)
-                    marketplaces = alias_orig_field.get('marketplaces')
-                    if marketplaces and (len(marketplaces) != 1 or marketplaces[0] != 'xsoar'):
-                        alias_orig_field['marketplaces'] = ['xsoar']
-                        click.secho(f'\n================= Updating file {alias_field_file_path} =================', fg='bright_blue')
-                        self.save_alias_field_file(dest_file_path=alias_field_file_path, field_data=alias_orig_field)
-
-    def get_incident_fields_as_dict_from_id_set(self):
-        incident_fields_dict = {}
-        if self.id_set_path:
-            id_set = open_id_set_file(self.id_set_path)
-            incident_field_list = id_set.get('IncidentFields')
-            for incident_field in incident_field_list:
-                field_id = list(incident_field.keys())[0]
-                incident_fields_dict[field_id] = incident_field[field_id]
-
-        return incident_fields_dict
-
-    def save_alias_field_file(self, dest_file_path, field_data):
+        if aliases:
+            aliased_fields = self._get_incident_fields_by_aliases(aliases)
+            for alias_field in aliased_fields:
+
+                alias_field_file_path = alias_field.pop('file_path', '')  # pop it as it's not part of the field and shouldn't be saved
+                marketplaces = get_item_marketplaces(item_path=alias_field_file_path, item_data=alias_field)
+
+                if len(marketplaces) != 1 or marketplaces[0] != 'xsoar':
+                    alias_field['marketplaces'] = ['xsoar']
+                    click.secho(f'\n================= Updating file {alias_field_file_path} =================', fg='bright_blue')
+                    self._save_alias_field_file(dest_file_path=alias_field_file_path, field_data=alias_field)
+
+    def _get_incident_fields_by_aliases(self, aliases: List[dict]) -> List[dict]:
+        """Get from the id_set the actual fields for the given aliases
+
+        Args:
+            aliases (list): The alias list.
+
+        Returns:
+            (list): the fileds as list of dicts.
+        """
+        res = []
+        alias_ids: set = {f'incident_{alias.get("cliName")}' for alias in aliases}
+        id_set = open_id_set_file(self.id_set_path)
+        incident_field_list: list = id_set.get('IncidentFields')
+
+        for incident_field in incident_field_list:
+            field_id = list(incident_field.keys())[0]
+            if field_id in alias_ids:
+                alias_data = incident_field[field_id]
+                alias_file_path = alias_data.get('file_path')
+                aliased_field, _ = get_dict_from_file(path=alias_file_path)
+                aliased_field['file_path'] = alias_file_path
+                res.append(aliased_field)
+        return res
```

Comment:
1. do not add the field's dict from the file to extract it afterward - return two values instead.
2. performance-wise: you are building the list here and iterating over it in the calling function - using `yield` will improve the run-time.
```suggestion
                yield aliased_field, alias_file_path
```

Reply:
Very nice!